pkg/collector/corechecks/gpu/nvmlmetrics/collector.go

```diff
@@ -21,9 +21,20 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+const (
+	tagVendor    = "gpu_vendor:nvidia"
+	tagNameModel = "gpu_model"
+	tagNameUUID  = "gpu_uuid"
+)
+
 // Collector is the main struct responsible for collecting metrics from the NVML library.
 type Collector struct {
-	lib        nvml.Interface
+	// lib is the NVML library interface
+	lib nvml.Interface
+
+	// collectors is a map of subsystems that can be used to collect metrics.
+	// Each device has its own list of collectors, as support for each collector
+	// might vary between devices
```

Comment:
can you also provide one example as part of this comment to show case where different gpu devices would have different sub-systems (and explain why that could happen)

Reply:
GPM is the example (and what motivated to add this), although I'm not really sure why GPM is not always supported, seems to be related to the hardware itself.